### PR TITLE
Feature/azure

### PR DESCRIPTION
**Pull request by chungyau97**
**Test Setup**

Azure ChatOpenAI
image

Azure OpenAI LLMs & Embeddings
image

### CODE REVIEW

File: packages/components/nodes/chatmodels/AzureChatOpenAI/AzureChatOpenAI.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        this.baseClasses = [this.type, ...getBaseClasses(ChatOpenAI)]
+        this.inputs = [
+            {
+                label: 'Azure OpenAI Api Key',
+                name: 'azureOpenAIApiKey',
+                type: 'password'
+            },
+            {
+                label: 'Model Name',
+                name: 'modelName',
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/AzureChatOpenAI/AzureChatOpenAI.ts:32]: not sure if these model names are correct. At least when checked docs - its called gpt-35-turbo. Refer to here: https://learn.microsoft.com/en-GB/azure/cognitive-services/openai/concepts/models

File: packages/components/nodes/llms/Azure OpenAI/AzureOpenAI.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        this.baseClasses = [this.type, ...getBaseClasses(OpenAI)]
+        this.inputs = [
+            {
+                label: 'Azure OpenAI Api Key',
+                name: 'azureOpenAIApiKey',
+                type: 'password'
+            },
+            {
+                label: 'Model Name',
+                name: 'modelName',
```
Comment (HenryHengZJ) [inline packages/components/nodes/llms/Azure OpenAI/AzureOpenAI.ts:31]: Refer to here for models: https://learn.microsoft.com/en-GB/azure/cognitive-services/openai/concepts/models

File: packages/components/nodes/embeddings/AzureOpenAIEmbedding/AzureOpenAIEmbedding.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                label: 'Azure OpenAI Api Key',
+                name: 'azureOpenAIApiKey',
+                type: 'password'
+            },
+            {
+                label: 'Model Name',
+                name: 'modelName',
+                type: 'options',
+                options: [
+                    {
```
Comment (HenryHengZJ) [inline packages/components/nodes/embeddings/AzureOpenAIEmbedding/AzureOpenAIEmbedding.ts:35]: We don't have model name is OpenAIEmbedding, think we can leave this out

File: packages/components/nodes/chatmodels/AzureChatOpenAI/AzureChatOpenAI.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                label: 'Azure OpenAI Api Deployment Name',
+                name: 'azureOpenAIApiDeploymentName',
+                type: 'string',
+                placeholder: 'YOUR-DEPLOYMENT-NAME'
+            },
+            {
+                label: 'Azure OpenAI Api Version',
+                name: 'azureOpenAIApiVersion',
+                type: 'string',
+                placeholder: 'YOUR-API-VERSION'
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/AzureChatOpenAI/AzureChatOpenAI.ts:74]: Can we change api version to options: 
image

File: packages/components/nodes/llms/Azure OpenAI/AzureOpenAI.ts [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                placeholder: 'YOUR-DEPLOYMENT-NAME'
+            },
+            {
+                label: 'Azure OpenAI Api Version',
+                name: 'azureOpenAIApiVersion',
+                type: 'string',
+                placeholder: 'YOUR-API-VERSION'
+            },
+            {
+                label: 'Azure OpenAI Api Embeddings Deployment Name',
```
Comment (HenryHengZJ) [inline packages/components/nodes/llms/Azure OpenAI/AzureOpenAI.ts:112]: Embeddings needed? 
image

File: packages/components/nodes/llms/Azure OpenAI/AzureOpenAI.ts [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+                name: 'azureOpenAIApiEmbeddingsDeploymentName',
+                type: 'string',
+                placeholder: 'YOUR-EMBEDDINGS-NAME',
+                optional: true,
+                additionalParams: true
+            },
+            {
+                label: 'Azure OpenAI Api Completions Deployment Name',
```
Comment (HenryHengZJ) [inline packages/components/nodes/llms/Azure OpenAI/AzureOpenAI.ts:120]: Not needed

File: packages/components/nodes/chatmodels/AzureChatOpenAI/AzureChatOpenAI.ts [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+            },
+            {
+                label: 'Temperature',
+                name: 'temperature',
+                type: 'number',
+                default: 0.9,
+                optional: true
+            },
+            {
+                label: 'Azure OpenAI Api Instance Name',
```
Comment (HenryHengZJ) [inline packages/components/nodes/chatmodels/AzureChatOpenAI/AzureChatOpenAI.ts:59]: Same, make sure just include fields that are needed: https://js.langchain.com/docs/modules/models/chat/integrations

File: packages/components/nodes/llms/Azure OpenAI/AzureOpenAI.ts (hunk 5 above)
Comment (chungyau97) [reply packages/components/nodes/llms/Azure OpenAI/AzureOpenAI.ts:112]: `azureOpenAIApiEmbeddingsDeploymentName` & `azureOpenAIApiCompletionsDeploymentName` is created under the `AzureOpenAIInput` interface, so I listed it additional parameter.

**Review by HenryHengZJ — APPROVED** [review]

**Pull request merged**